content/v2.0/query-data/guides/scalar-values.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Extract scalar values in Flux
+description: >
+  placeholder
+menu:
+  v2_0:
+    name: Extract scalar values
+    parent:  How-to guides
+weight: 210
+v2.0/tags: [scalar]
+---
+
+Use Flux [stream and table functions](/v2.0/reference/flux/stdlib/built-in/transformations/stream-table/)
+to extract scalar values from Flux query output for things like dynamically setting
+variables using query results.
+
+**To extract scalar values from output:**
+
+1. [Extract a table](#extract-a-table)
+2. Use the extracted table and [extract an array of column values](#extract-a-column-from-the-table)
+   or [extract a row as an object](#extract-a-row-from-the-table).
+
+_The samples on this page use the [sample data provided below](#sample-data)._
+
+{{% warn %}}
+#### Current limitations
+- The InfluxDB user interface (UI) does not support raw scalar output, but you can
+  use [`map()`](/v2.0/reference/flux/stdlib/built-in/transformations/map/) to add
+  scalar values to data.
+- The [Flux REPL ](/v2.0/reference/cli/influx/repl) does not currently support
+  Flux stream and table functions (also known as "dynamic queries").
+  See [#15321](https://github.com/influxdata/influxdb/issues/15231).
+
+{{% /warn %}}
+
+## Extract a table
+Flux formats query results as a stream of tables.
+To extract a scalar value, first first use
+[`tableFind()`](/v2.0/reference/flux/stdlib/built-in/transformations/stream-table/tablefind/)
+to extract a single table from the stream of tables.
+
+{{% note %}}
+If query results include only one table, it is still formatted as a stream of tables.
+{{% /note %}}
+
+The `tableFind()` extracts the **first** table in a stream of tables whose
+[group key](/v2.0/reference/glossary/#group-key) values match the
+[predicate function](/v2.0/reference/glossary/#predicate-expression) defined by
+the `fn` parameter.
+The predicate function expects a `key` object, which represents the group key of
+each table.
+
+```js
+sampleData
+  |> tableFind(fn: (key) =>
+      key._field == "temp" and
+      key.location == "sfo"
+  )
+```
+
+This returns a single table:
+
+| _time                | location | _field | _value |
+|:-----                |:--------:|:------:| ------:|
+| 2019-11-01T12:00:00Z | sfo      | temp   | 65.1   |
+| 2019-11-01T13:00:00Z | sfo      | temp   | 66.2   |
+| 2019-11-01T14:00:00Z | sfo      | temp   | 66.3   |
+| 2019-11-01T15:00:00Z | sfo      | temp   | 66.8   |
+
+{{% note %}}
+#### Minimize the number of input tables
+Flux functions do not guarantee the order in which tables are returned.
+`tableFind()` simply extracts the first table that matches the `fn` predicate.
```

Comment:
does tableFind() always return one table? Curious bc otherwise, why would the table order matter?

Reply:
Yes, `tableFind()` only ever returns one table – the **first** table it finds that matches the predicate. For example, using the sample data in the guide, you have tables with the following group keys:

```
[location="sfo", _field="temp"]
[location="kjfk", _field="temp"]
[location="kord", _field="temp"]
```

There's no guarantee that the tables will come in that order. So if you're using the predicate:

```
(key) => key._field == "temp"
```

You'll get temp data from one of the three locations, but there's no guarantee as to which one.

I think I'll update this note a bit. You either need to minimize the number of tables going into `tableFind()` or be very specific in your predicate function.